### PR TITLE
Enable use of splines of degree 0

### PR DESCRIPTION
**Pull request by e-moral-sanchez**
Compute Greville points and derivatives of splines of degree 0. Activate all unit tests for degree 0. This enables the use of splines of degree 1 for the first space in a De Rham sequence.

### CODE REVIEW

File: psydac/core/bsplines.py [hunk 1]
```diff
@@ -597,7 +597,7 @@ def make_knots(breaks, degree, periodic, multiplicity=1, out=None):
     # Consistency checks
     assert len(breaks) > 1
     assert all( np.diff(breaks) > 0 )
-    assert degree > 0
+    #assert degree > 0
```
Comment (yguclu) [inline psydac/core/bsplines.py:600]: Maybe change to this?
```python
assert degree >= 0
```

File: psydac/core/bsplines_pyccel.py [hunk 2]
```diff
@@ -440,7 +445,7 @@ def collocation_matrix_p(knots: 'float[:]', degree: int, periodic: bool, normali
                 for j in range(degree + 1):
                     actual_j = (spans[i] - degree + j) % nb
                     out[i, actual_j] = basis[i, j] * scaling[spans[i] - degree + j]
-
+
```
Comment (yguclu) [inline psydac/core/bsplines_pyccel.py:448]: Useless whitespace change

**Review by yguclu — CHANGES_REQUESTED** [review]

File: psydac/core/bsplines_pyccel.py (hunk 2 above)
Comment (e-moral-sanchez) [reply psydac/core/bsplines_pyccel.py]: It is fixed

File: psydac/core/bsplines.py (hunk 1 above)
Comment (e-moral-sanchez) [reply psydac/core/bsplines.py]: It is fixed

**Review by yguclu — APPROVED** [review]

**Review by yguclu — APPROVED** [review]

**Pull request merged**